src/framework/actions/index.ts

```diff
@@ -22,6 +25,10 @@ export const performAction = (
   case "PLAY_CARD":
     playCard($, action.cardID, action.boost);
     break;
+  case "TECH":
+    action.cards = action.cards.slice(0, 2);
```

Comment:
Rather than mutating the input here, I think we should pass it into the impl as-is and throw an error if we get an unexpected number of cards (too many e.g.)

Reply:
I wasn't sure whether it would be better to do the guarding here at the action level or in the processing function itself. There are other things we need to guard against as well (created a card for them), so there are several reasons why a particular tech attempt could fail. I'll make it a failure state like you suggested..